### ISSUE TITLE
Missing dispose() on a controller

### ISSUE THREAD

**Issue by simon-vorac** [xuzhongpeng/qrcode_flutter#5]
Hi, could you please expose dispose() method on the **QRCaptureController**?

**Comment by joaomagfreitas** [comment]
@simon-vorac Let me know if this works for you

https://github.com/xuzhongpeng/qrcode_flutter/pull/6

**Comment by xuzhongpeng** [comment]
> Hi, could you please expose dispose() method on the **QRCaptureController**?

@simon-vorac @freitzzz  Sorry. We can't expose `dispose` on the QRCaptureController. If you want to `disposed` the `barcodeView`. You can use `pause` instead.

The `barcodeView` and the PlatformView are closely related. It will `disposed` when we close PlatformView.If we disposed the `barcodeView` before `PlatforView` closed, there will be throw error.

Can you tell me why you want to expose `dispose` method?

**Comment by simon-vorac** [comment]
Hi, sorry for late reply. I understand and if it is handled by the `PlatformView`, then I am ok with it. I'll be using `pause`.
It is a controller and I am used to disposing of any controller after it's use. Anyway, thank you for your time to answer and checking the issue.

**Comment by Holofox** [comment]
@xuzhongpeng, there is one moment when we open the camera, quickly close it and try to open it again, we get the following exception:

```console
D/CameraPreview(19624): resume()
D/CameraInstance(19624): Opening camera
W/CameraBase(19624): An error occurred while connecting to camera 0: Status(-8, EX_SERVICE_SPECIFIC): '6: connectHelper:1718: Camera "0" disabled by policy'
E/CameraInstance(19624): Failed to open camera
E/CameraInstance(19624): java.lang.RuntimeException: Fail to connect to camera service
E/CameraInstance(19624): 	at android.hardware.Camera.<init>(Camera.java:557)
E/CameraInstance(19624): 	at android.hardware.Camera.open(Camera.java:402)
E/CameraInstance(19624): 	at com.google.zxing.client.android.camera.open.OpenCameraInterface.open(OpenCameraInterface.java:83)
E/CameraInstance(19624): 	at com.journeyapps.barcodescanner.camera.CameraManager.open(CameraManager.java:138)
E/CameraInstance(19624): 	at com.journeyapps.barcodescanner.camera.CameraInstance$3.run(CameraInstance.java:190)
E/CameraInstance(19624): 	at android.os.Handler.handleCallback(Handler.java:907)
E/CameraInstance(19624): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/CameraInstance(19624): 	at android.os.Looper.loop(Looper.java:223)
E/CameraInstance(19624): 	at android.os.HandlerThread.run(HandlerThread.java:67)
D/CameraPreview(19624): pause()
D/CameraInstance(19624): Closing camera
```

But if you call the `dispose` method, this no longer happens.

**Comment by xuzhongpeng** [comment]
fix in 2.0.0
